Adf4350 remove of use

tachicialex opened the pull request:
This patch replaces functions from of.h and of_gpio.h with their counterparts 
from property.h and gpio/consumer.h.

commodo commented on drivers/iio/frequency/adf4350.c:
```diff
@@ -542,29 +536,26 @@ static int adf4350_probe(struct spi_device *spi)
 				goto error_disable_reg;
 			}
 		}
+	} else {
```
this changes behavior;
before, if there was no gpio defined, it would not error out on probe;

now it seems to do;

commodo commented on drivers/iio/frequency/adf4350.c:
```diff
@@ -382,11 +379,8 @@ static struct adf4350_platform_data *adf4350_parse_dt(struct device *dev)
-	else
-		pdata->gpio_lock_detect = ret;
+	pdata->gpio_lock_detect = devm_gpiod_get_optional(dev, "lock_detect",
+							  GPIOD_IN);
```
maybe do an `if (IS_ERR(pdata->gpio_lock_detect))  return PTR_ERR(pdata->gpio_lock_detect);` here;

if there is no GPIO, all is fine; we get a NULL and `IS_ERR(pdata->gpio_lock_detect)` does not evaluate true;
if we get an error, we would want to fail he init code here;

tachicialex replied:
i must return a pointer here that contains the error code:

```
if (IS_ERR(pdata->gpio_lock_detect))
		return (void *)pdata->gpio_lock_detect;
```
Is this ok?

nunojsa replied:
It seems you could use `ERR_CAST(pdata->gpio_lock_detect)` here...

commodo commented on drivers/iio/frequency/adf4350.c:
```diff
@@ -367,91 +367,92 @@ static struct adf4350_platform_data *adf4350_parse_dt(struct device *dev)
 
 	strncpy(&pdata->name[0], np->name, SPI_NAME_SIZE - 1);
 
-	tmp = 10000;
```
this patch doesn't remove `of.h` yet;
it just replaces all `of_property_read_u32()` with `device_property_read_u32()`
so, maybe the patch should be named a bit more like `iio: frequency: adf4350: replace of_property* functions with device_property functions`

to completely remove of.h, you need to:
1. remove `struct device_node *np = dev->of_node;` from this file
2. remove `#ifdef CONFIG_OF`
3. remove the of.h include

regarding point 2, maybe it also makes sense to remove `static struct adf4350_platform_data default_pdata` ; and make sure that `adf4350_parse_dt()` initializes the same data as in `default_pdata`

maybe it also makes sense to move the contents of `linux/iio/frequency/adf4350.h` into `adf4350.c`; they should not have been put there, since it doesn't make sense; those definitions are private to the driver;

commodo replied:
otherwise, this patch is fine on it's own [from a content perspective]

commodo commented on drivers/iio/frequency/adf4350.c:
```diff
@@ -478,6 +478,8 @@ static int adf4350_probe(struct spi_device *spi)
 		pdata = adf4350_parse_dt(&spi->dev);
 		if (pdata == NULL)
```
there is no error-path returning NULL in `adf4350_parse_dt()`, so you can probably remove this NULL check

commodo commented on drivers/iio/frequency/adf4350.c:
```diff
@@ -357,103 +356,103 @@ static const struct iio_info adf4350_info = {
 
-	tmp = 0;
-	of_property_read_u32(np, "adi,reference-div-factor", &tmp);
+	if (device_property_read_u32(dev, "adi,reference-div-factor", &tmp))
```
i liked the previous version of this patch that does the `of_property_read_u32()` -> `device_property_read_u32` conversion;

removing `of.h` is still incomplete here; and this code still needs DT;
you'd still need to remove `#ifdef CONFIG_OF` and related parts

tachicialex commented:
Changes V3:
- removed use of of.h
- added another commit which adds error codes in parse_dt() function
- removed else branch at gpio in probe at condition: `if (pdata->gpio_lock_detect) `

commodo commented on drivers/iio/frequency/adf4350.c:
```diff
@@ -354,117 +353,109 @@ static const struct iio_info adf4350_info = {
 		return ERR_PTR(-ENOMEM);
 
-	strncpy(&pdata->name[0], np->name, SPI_NAME_SIZE - 1);
+	strncpy(&pdata->name[0], dev_name(dev), SPI_NAME_SIZE - 1);
```
just as a note, this changes the name of the device; which isn't a big problem;
at some point [later] we may want to remove the `name` field from the `adf4350_platform_data` struct
it doesn't add any value
